server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -194,6 +197,48 @@ public void testCancelPrimaryAllocation() throws Exception {
         assertSegmentStats(REPLICA_COUNT);
     }
 
+    /**
+     * This test adds a new replica shard to an existing cluster which already has few docs inserted before adding replica.
+     * We don't perform any refresh on index and assert new replica shard on doc hit count.
+     * This test makes sure that when a new replica is added to an existing cluster it gets all latest segments from primary even without a refresh.
+     */
+    public void testAddNewReplica() throws Exception {
```

Comment:
This test is very similar to `testStartReplicaAfterPrimaryIndexesDocs`, can we reuse that test?  That test currently indexes a doc after the replica is recovered to force another round of replication, but you could assert the doc count is sync'd on line 412 after ensureGreen().

Reply:
Yes, I think you right. Let me see if we can reuse it